nnvm/tests/python/compiler/test_top_level1.py

```diff
@@ -64,8 +22,8 @@ def backward(head_grads, x):
 
     dtype = "float32"
     dshape = (1, 3, 32, 32)
-    inputs = [('x', dshape, x)]
-    helper(y, inputs, dtype, forward, backward)
+    inputs = [(x, dshape)]
+    check_function(y, inputs, forward, backward, dtype=dtype)
```

Comment:
Here we are setting grad_input_vars=inputs, but inputs are actually input_shapes. It would be more clear if we can refactor the interface of check_function.